### DISCUSSION TITLE
Foreign key error when deleting object

### DISCUSSION THREAD

**Discussion by tika** [prisma/prisma#10911] [Q&A] [answered]
hia, whenever I try to delete a set or user, I am greeted with the kind error that is below.
![image](https://user-images.githubusercontent.com/48658947/147821958-1b643838-b4f3-4bca-b7c6-f5299bcc5919.png)

any help would be appreciated.

here's my schema:
```prisma
model User {
  id           Int      @id @default(autoincrement())
  createdAt    DateTime @default(now())
  name         String
  email        String   @unique
  password     String
  avatar       String?
  sets         Set[]    @relation(name: "userInSet")
  authoredSets Set[]    @relation(name: "userAuthorsSet")
}

model Set {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())
  name      String
  terms     Term[]
  users     User[]   @relation(name: "userInSet")
  author    User     @relation(name: "userAuthorsSet", fields: [authorId], references: [id])
  authorId  Int
}

model Term {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())
  pinyin    String
  english   String
  chinese   String
  set       Set      @relation(fields: [setId], references: [id])
  setId     String
}
```

**Comment by ajmnz** [comment, marked as answer]
I believe that Prisma doesn't let you delete a `Set` because the relation between `Term` and `Set` is not optional. Deleting a `Set` that has multiple related `Term`s would cause those to have `setId` as `null`, which is something your schema doesn't allow.
The same situation applies to the relation between `User` and `Set`.

To solve this and be able to delete the records without getting the error, either make the relations optional or take a look at [referential actions](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions) in the docs.

**Reply by tika** [reply]
Thank you very for the new information, I did not know about this feature - you're a life saver! 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
